src/CompanyService.Business/Commands/Office/FindOfficesCommand.cs

```diff
@@ -15,47 +19,51 @@ public class FindOfficesCommand : IFindOfficesCommand
     private readonly IOfficeRepository _officeRepository;
     private readonly IOfficeInfoMapper _mapper;
     private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IBaseFindRequestValidator _baseFindValidator;
+
 
     public FindOfficesCommand(
-        IOfficeRepository officeRepository,
-        IOfficeInfoMapper mapper,
-        IHttpContextAccessor httpContextAccessor)
+      IOfficeRepository officeRepository,
+      IOfficeInfoMapper mapper,
+      IHttpContextAccessor httpContextAccessor,
+      IBaseFindRequestValidator baseFindValidator)
     {
       _officeRepository = officeRepository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _baseFindValidator = baseFindValidator;
     }
 
-    public FindResultResponse<OfficeInfo> Execute(int skipCount, int takeCount, bool? includeDeactivated)
+    public FindResultResponse<OfficeInfo> Execute(BaseFindFilter filter)
     {
-      if (skipCount < 0)
+      FindResultResponse<OfficeInfo> response = new();
+
+      if (!_baseFindValidator.ValidateCustom(filter, out List<string> errors))
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
-        return new()
-        {
-          Status = OperationResultStatusType.Failed,
-          Errors = new() { "Skip count can't be less than 0." }
-        };
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors = errors;
+        return response;
       }
 
-      if (takeCount < 1)
+      response.Body = _officeRepository
+        .Find(filter, out int totalCount)
+        .Select(_mapper.Map)
+        .ToList();
+
+      response.TotalCount = totalCount;
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      if (response.Body == null)
       {
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
```

Comment:
we don't find nothing, but why it is error? why we don't return empty list

Reply:
done